src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -455,6 +458,32 @@ public void updateSettings(Settings settings) {
     }
 
     /**
+     * Adds an custom executor into the thread pool framework. It is up to the caller to provide a valid executor
+     * instance that is in a started state.
+     *
+     * This is useful in the case thread pools created by plugins that want their thread pools be included in the stats
+     * and info apis.
+     */
+    public void addCustomExecutor(Executor executor, Info info) {
```

Comment:
@martijnvg I think we should change the signature here and ask for something like this:

```Java
public void addCustomExecutor(String name, Settings settings) {
  ExecutorHolder holder = rebuild(name, null, settings, defaultSettings);
  ....
}

```

makes sense?